app/javascript/components/rooms/FeatureTabs.jsx

```diff
@@ -4,20 +4,30 @@ import RoomRecordings from '../recordings/RoomRecordings';
 import Presentation from './Presentation';
 import SharedAccess from '../shared_accesses/SharedAccess';
 import RoomSettings from '../room_settings/RoomSettings';
+import useSiteSetting from '../../hooks/queries/site_settings/useSiteSetting';
 
 export default function FeatureTabs() {
+  const { data: shareRoomEnabled } = useSiteSetting('ShareRooms');
+  const { data: preuploadEnabled } = useSiteSetting('PreuploadPresentation');
+
   return (
     <Row className="pt-5 mx-0">
       <Tabs defaultActiveKey="recordings" unmountOnExit>
         <Tab eventKey="recordings" title="Recordings">
           <RoomRecordings />
         </Tab>
-        <Tab eventKey="presentation" title="Presentation">
-          <Presentation />
-        </Tab>
-        <Tab eventKey="access" title="Access">
-          <SharedAccess />
-        </Tab>
+        { preuploadEnabled
+          && (
+          <Tab eventKey="presentation" title="Presentation">
+            <Presentation />
+          </Tab>
+          )}
+        { shareRoomEnabled
+          && (
+          <Tab eventKey="access" title="Access">
+            <SharedAccess />
+          </Tab>
+          )}
```

Comment:
This is an example of how it could be used in the front-end

Reply:
I think that we can approach this differently without having to make a full request for each setting each time we visit a room.

Reply:
IMO these settings should be part of the server state given by the `useSessions`, whcih will cache these configs in the `queryCache` and `react-query` will make sure to keep it in sync.

The `AuthProvider` or a new concerned provider will then fetch these configs for us to access when needed through its context.

The user role, its permissions and the site settings (like theme ...) designs a server state also.

Or we can make an endpoint that fetches them in a single request.